core/model/src/main/java/org/eclipse/sw360/antenna/model/artifact/ArtifactCoordinates.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2018.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.sw360.antenna.model.artifact;
+
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactIdentifier;
+import org.eclipse.sw360.antenna.model.coordinates.Coordinate;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public final class ArtifactCoordinates implements ArtifactIdentifier<ArtifactCoordinates>, ArtifactFact<ArtifactCoordinates> {
+    private final Map<String, Coordinate> coordinates = new HashMap<>();
+
+    private void putCoordinate(Coordinate coordinate) {
+        if (coordinate != null && ! coordinate.isEmpty()) {
+            coordinates.put(coordinate.getType(), coordinate);
+        }
+    }
+
+    private void putCoordinates(Collection<Coordinate> coordinates) {
+        coordinates.forEach(this::putCoordinate);
+    }
+
+    public ArtifactCoordinates(String name, String version) {
+        final Coordinate coordinate = Coordinate.builder()
+                .withName(name)
+                .withVersion(version)
+                .withType(Coordinate.Types.GENERIC)
+                .build();
+        coordinates.put(coordinate.getType(), coordinate);
+    }
+
+    public ArtifactCoordinates(Coordinate... coordinatesToAdd) {
+        putCoordinates(Arrays.asList(coordinatesToAdd));
+    }
+
+    public ArtifactCoordinates(String... coordinateStringsToAdd) {
+        for (String coordinateStringToAdd: coordinateStringsToAdd) {
+            putCoordinate(Coordinate.builder(coordinateStringToAdd).build());
+        }
+    }
+
+    public ArtifactCoordinates(Collection<Coordinate> coordinatesToAdd) {
+        putCoordinates(coordinatesToAdd);
+    }
+
+    public ArtifactCoordinates(Set<String> coordinateStringsToAdd) {
+        for (String coordinateStringToAdd: coordinateStringsToAdd) {
+            putCoordinate(Coordinate.of(coordinateStringToAdd));
+        }
+    }
+
+    public boolean containsPurl(String coordinateString)  {
+        return containsPurl(Coordinate.of(coordinateString));
+    }
+
+    public boolean containsPurl(Coordinate coordinate) {
+        return coordinates.values().contains(coordinate);
+    }
+
+    public Set<Coordinate> getCoordinates() {
```

Comment:
There we are again in the discussion whether Collection is not the better result. Actually in such a case an unmodifiable Collection seems to represent the information given back the best, IMO

Reply:
`Collection` is the better input. But specific types are the better result.

If a function understands `Collection` as **input, one can understand everything**. If one outputs a `Collection`, **the caller must understand all types of collections**. If I return a Set the user still can treat it as a Collection, so nothing is lost here.